packages/std/src/math/decimal.rs

```diff
@@ -53,6 +55,31 @@ impl Decimal {
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
+
+    /// Returns the multiplicative inverse `1/d` for decimal `d`.
+    ///
+    /// If `d` is zero, none is returned.
+    pub fn inv(&self) -> Option<Decimal> {
+        if self.is_zero() {
+            None
+        } else {
+            // Let self be p/q with p = self.0 and q = DECIMAL_FRACTIONAL_SQUARED.
+            // Now we calculate the inverse a/b = q/p such that b = DECIMAL_FRACTIONAL_SQUARED.
+            Some(Decimal(DECIMAL_FRACTIONAL_SQUARED / self.0))
+        }
+    }
+}
+
+impl Fraction<u128> for Decimal {
```

Comment:
Not yet, but I would like to encourage the usage of fractions for more advanced math. Rational numbers (i.e. numbers that can be represented as a fraction) make up a huge part of the numbers we need in applications and all operations on rational numbers can be reduced to integer math. Decimal (i.e. a fraction with a constant denominator) is just the beginning.

Reply:
The Fraction trait now contains `inv` as well. I.e. every fraction is required to have it. This makes sense because for every `p/q` you can calculate `q/p` one way or another.